core/src/main/scala/za/co/absa/fadb/DBFunction.scala

```diff
@@ -206,4 +222,85 @@ object DBFunction {
      */
     def apply(values: I)(implicit me: MonadError[F, Throwable]): F[Option[R]] = optionalResult(values)
   }
+
+  /**
+    *  `DBMultipleResultFunctionWithStatus` is an abstract class that represents a database function returning
+    *  multiple results with status information.
+    *  It extends the [[DBFunctionWithStatus]] class and overrides the apply method to return a sequence of results.
+    */
+  abstract class DBMultipleResultFunctionWithStatus[I, R, E <: DBEngine[F], F[_]](
+    functionNameOverride: Option[String] = None
+  )(implicit schema: DBSchema, dBEngine: E)
+    extends DBFunctionWithStatus[I, R, E, F](functionNameOverride)
+      with StatusAggregator {
```

Comment:
I don't like the fact it requires `StatusAggregator` to be mixed-in. I believe this should be optional and the return value of `DBMultipleResultFunctionWithStatus` should be a sequence of `Either` (or it's alias). The aggregation should be in my opinion applied optionally later on as one might be interested also in all statuses.

Reply:
yes that's basically the hardest 'design decision' in this PR I think. I appreciate / see the benefits of both approaches, that's why I was kind of undecided yet and 'waiting' for 3rd opinion :D

Reply:
Okay I though about it and:
* I think status aggregation in general is gonna be used quite common and is therefore very helpful - just need to provide implementation that is convenient for us and users
* I still really like and prefer mix-in approach - it's consistent and 'expected' from user point of view how things are plugged together, like nice reusable and changeable lego blocks
* I understand that having a mandatory agg implementation might be too much and sometimes not needed. Therefore I implemented it separately - one with mixin and the other without.
* With this 'new' approach, I created respective 'utils' ('helper' classes) to both, Doobie and Slick, and sufficiently covered it all with tests.

Does that influence your opinion on this status aggregation related feature?